internal/pkg/service/buffer/worker/service/file_swap.go

```diff
@@ -34,48 +34,41 @@ func (s *Service) swapFile(fileKey key.FileKey, reason string) (err error) {
 		Context: func() (context.Context, context.CancelFunc) {
 			return context.WithTimeout(context.Background(), time.Minute)
 		},
-		Operation: func(ctx context.Context, logger log.Logger) task.Result {
+		Operation: func(ctx context.Context, logger log.Logger) (result task.Result) {
 			logger.Infof(`closing file "%s": %s`, fileKey, reason)
 
-			err := func() (err error) {
-				rb := rollback.New(logger)
-				defer rb.InvokeIfErr(ctx, &err)
+			rb := rollback.New(logger)
+			defer rb.InvokeIfErr(ctx, &result.Error)
```

Comment:
A tu je dalsi priklad, ked nam zmeny v `task.Result` spravia kod jednoduchsim.

Tu sme mali cely kod este obaleny do anonymnej funkcie `func() (err error) {`, pretoze ak operacia zlyha, tak robime rollback (volame opacne operacie nad Storage API, a v LIFO poradi).

Teraz to viem zapisat ako `defer rb.InvokeIfErr(ctx, &result.Error)`.

-----

Predtym sa mi to nechcelo prerabat, pre jeden use-case, ale teraz by som to takto musel mat skoro v kazdom tasku.